x/exchange/msg.go

```diff
@@ -0,0 +1,464 @@
+		errs = append(errs, err)
+	}
+
+	inBoth := IntersectionUint64(m.AskOrderIds, m.BidOrderIds)
+	if len(inBoth) > 0 {
+		errs = append(errs, fmt.Errorf("order ids duplicated as both bid and ask: %v", inBoth))
+	}
+
+	// Nothing to validate now for the ExpectPartial flag.
+
+	return errors.Join(errs...)
+}
+
+func (m MsgMarketSettleRequest) GetSigners() []sdk.AccAddress {
+	addr := sdk.MustAccAddressFromBech32(m.Admin)
+	return []sdk.AccAddress{addr}
+}
+
+func (m MsgMarketWithdrawRequest) ValidateBasic() error {
+	var errs []error
+
+	if _, err := sdk.AccAddressFromBech32(m.Admin); err != nil {
+		errs = append(errs, fmt.Errorf("invalid administrator %q: %w", m.Admin, err))
+	}
+
+	if m.MarketId == 0 {
+		errs = append(errs, errors.New("invalid market id: cannot be zero"))
+	}
+
+	if _, err := sdk.AccAddressFromBech32(m.ToAddress); err != nil {
+		errs = append(errs, fmt.Errorf("invalid to address %q: %w", m.ToAddress, err))
+	}
+
+	if err := m.Amount.Validate(); err != nil {
+		errs = append(errs, fmt.Errorf("invalid amount %q: %w", m.Amount, err))
+	} else if m.Amount.IsZero() {
+		errs = append(errs, fmt.Errorf("invalid amount %q: cannot be zero", m.Amount))
+	}
+
+	return errors.Join(errs...)
+}
+
+func (m MsgMarketWithdrawRequest) GetSigners() []sdk.AccAddress {
+	addr := sdk.MustAccAddressFromBech32(m.Admin)
+	return []sdk.AccAddress{addr}
+}
+
+func (m MsgMarketUpdateDetailsRequest) ValidateBasic() error {
+	var errs []error
+
+	if _, err := sdk.AccAddressFromBech32(m.Admin); err != nil {
+		errs = append(errs, fmt.Errorf("invalid administrator %q: %w", m.Admin, err))
+	}
+
+	if m.MarketId == 0 {
+		errs = append(errs, fmt.Errorf("invalid market id: cannot be zero"))
+	}
+
+	if err := m.MarketDetails.Validate(); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errors.Join(errs...)
+}
+
+func (m MsgMarketUpdateDetailsRequest) GetSigners() []sdk.AccAddress {
+	addr := sdk.MustAccAddressFromBech32(m.Admin)
+	return []sdk.AccAddress{addr}
+}
+
+func (m MsgMarketUpdateEnabledRequest) ValidateBasic() error {
+	var errs []error
+
+	if _, err := sdk.AccAddressFromBech32(m.Admin); err != nil {
+		errs = append(errs, fmt.Errorf("invalid administrator %q: %w", m.Admin, err))
+	}
+
+	if m.MarketId == 0 {
+		errs = append(errs, fmt.Errorf("invalid market id: cannot be zero"))
+	}
+
+	// Nothing to validate for the AcceptingOrders field.
+
+	return errors.Join(errs...)
+}
+
+func (m MsgMarketUpdateEnabledRequest) GetSigners() []sdk.AccAddress {
+	addr := sdk.MustAccAddressFromBech32(m.Admin)
+	return []sdk.AccAddress{addr}
+}
+
+func (m MsgMarketUpdateUserSettleRequest) ValidateBasic() error {
+	var errs []error
+
+	if _, err := sdk.AccAddressFromBech32(m.Admin); err != nil {
+		errs = append(errs, fmt.Errorf("invalid administrator %q: %w", m.Admin, err))
+	}
+
+	if m.MarketId == 0 {
+		errs = append(errs, fmt.Errorf("invalid market id: cannot be zero"))
+	}
+
+	// Nothing to validate for the AllowUserSettlement field.
```

Comment:
Probably don't need this comment.

Reply:
Here too, this tells the next person that that field wasn't forgotten, it's just that there's nothing to do for it.